### ISSUE TITLE
[Feature Request] Code style rule to allow dedented comments

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn#50631]
**Brief description:**

Please allow us to use dedented comments, without triggering `IDE0055`. This should be a new [formatting rule](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules).

**Languages applicable:**

c#

**Code example that the analyzer should report:**

This is currently allowed:

````csharp
  if (true) {
    fooBar();
    doSomething();
    //andDoAnotherThing();
    alsoThisStuff();
  }
````

But this fails due to `IDE0055` ("Fix formatting"):

````csharp
  if (true) {
    fooBar();
    doSomething();
  //andDoAnotherThing();     <-- dedent causes an error
    alsoThisStuff();
  }
````

**Additional information:**

This sort of commenting is quite common in various codebases, and other frameworks. Why it's desirable:
- it's MUCH neater when quickly scanning through the code - all the code is "in the right place", without being shifted to the right; also the IDE colors the comments (typically green) to make it easy to differentiate
- it plays nicely with source control: commenting / uncommenting code doesn't result in ugly diffs - so we can scan through more code more quickly

Please give us an option to allow this?

**Comment by CyrusNajmabadi** [comment]
Closing as not planned due to lack of broad feedback.
